### ISSUE TITLE
Should $modules->get("ModuleName") install a module?

### ISSUE THREAD

**Issue by adrianbj** [processwire/processwire-requests#135]
### Short description of the enhancement

This is a bit of a discussion about the behavior of `$modules->get()`

Currently,
```
$modules->get("ModuleName");
```
will install a module that is currently uninstalled. 
I think instead it should return an message/error if the requested module is not installed. 

To install a module you should have to use:

```
$modules->install("ModuleName");
```

### Why would the enhancement be useful to users?

Semantically I think it would be more appropriate to require the use of `install()`, but also I have come across many situations where I am testing and uninstall a module only to find it has been re-installed by a template file or another module requesting it via `get()`.

Of course this is a breaking change so might have to be saved for PW 4, but maybe it could be deprecated in 3.1?

Any thoughts?

**Comment by ghost** [comment]
How about a second, optional, boolean parameter passed to the get method?
Default is true so it will behave the same as now and when it is false the module will not be installed.

**Comment by netcarver** [comment]
The current side-effect of the get() method is not intuitive, but it is [documented](https://processwire.com/api/ref/modules/get/). Just using ```$modules->isInstalled('SomeModule');``` seems to work very well but I did have to google to find that out, even though that method is listed in the class documentation too. Maybe just a pointer in the right direction in the documentation for the ```Get()``` method is what is needed here.

**Comment by adrianbj** [comment]
@netcarver, thanks for your feedback. Despite being documented, I still don't think it makes sense when we have an `install()` method. I'd rather it was necessary to explicitly install first before `get()` can return anything. I just think the current behavior makes it too easy to have unintended consequences, but maybe it's just me who thinks this way?

**Comment by matjazpotocnik** [comment]
adrainbj, I have to agree with netcarver. It is unintuitive that get() installs the module and it should be the way you suggest, but it works that way for a long time and changing this might cause troubles. We have `getModule($name, array('noInstall' => true));`

**Comment by adrianbj** [comment]
> but it works that way for a long time and changing this might cause troubles

I totally agree, that is why I was suggesting deprecation and removal in PW 4.

It's certainly not the end of the world - just a thought for improving things.

Anyway, we'll see what Ryan thinks :)

Thanks both of you for your thoughts!
